augur/db/export/__init__.py

```diff
@@ -0,0 +1,19 @@
+"""
+Export data from a database.
+"""
+from augur.utils import first_line
+from ..load import DEFAULT_LOADED_METADATA_TABLE
+from .sqlite3 import export as export_sqlite3
+
+
+def register_parser(parent_subparsers):
+    parser = parent_subparsers.add_parser("export", help=first_line(__doc__))
+    parser.add_argument('--sqlite3', required=True, help="SQLite3 database file to export from.")
```

Comment:
> Or is the idea that we would add other arguments to support other implementations like `--postgres-server`?

Yes, this was my intention of naming it specifically `--sqlite3`. There are potential use cases for `augur db` with other types of databases, and the code here has been organized with that in mind.

However, I suspect much of our initial use will be with SQLite3 – passing the database file around as currently done with sequences and metadata files. Even if this means users might be unfamiliar with SQLite3 and do not necessarily need to know how it works, knowing what it is under the hood can be beneficial for debugging and general awareness.

For the `--database` idea specifically: I suppose we could use `pyodbc` and accept an ODBC connection string, but that's not so user-friendly (at least for SQLite3 which requires downloading an [ODBC driver](http://www.ch-werner.de/sqliteodbc/) separately):

- `--database "DRIVER={SQLite3 ODBC Driver};SERVER=localhost;DATABASE=example.db;Trusted_connection=yes"`
- `--database "DRIVER={PostgreSQL ODBC Driver(UNICODE)};SERVER=some.postgresql.endpoint;PORT=5432;DATABASE=example;UID=user;PWD=pass"`

My intention is a more user-oriented CLI that allows for custom "connection strings" per database type:

- `--sqlite3 example.db`
- `--postgresql "postgresql://user:pass@some.postgresql.endpoint:5432/example"`

Reply:
Actually, we could support custom "connection values" and determine database type internally.

- `--database example.db`
- `--database "postgresql://user:pass@some.postgresql.endpoint:5432/example"`

Under the hood:

```
if is_sqlite_file(args.database):
    load_sqlite3(…)
elif is_postgresql_connection_string(args.database):
    load_postgres(…)
…
else:
    raise AugurError("Unsupported connection value. Run augur db load --help to for supported options.")
```

I'd be open to this alternative. Does it sound better to you?

Reply:
Sounds better to me. It's common convention to have a single argument/option/parameter/etc. (here, `--database`) that takes some sort of [DSN (data source name)](https://en.wikipedia.org/wiki/Data_source_name) which identifies the implementation. The `postgresql://` URLs and ODBC connection strings are both examples of DSNs.

Reply:
Let's not reinvent the wheel though, so we should pick an existing DSN convention and stick with it. Some prior art:

- https://github.com/libwww-perl/URI-db ([background](https://justatheory.com/2013/11/toward-a-database-uri-standard/))
- https://github.com/xo/dburl

The first link also includes other references at the bottom of its README.

Reply:
Relatedly, let's make `--output-sqlite3` → `--output-database` and so on for similar options.

(Should we use `db` instead of `database`, e.g. `--db` and `--output-db`, so it matches with `augur db`? Or inversely, should we rename `augur db` to `augur database` and leave the options as `--database`? Maybe we should just alias 'em all.)

Reply:
Good suggestion on aliasing. Renamed `--sqlite3` → `--db` and `--output-sqlite3` → `--output-db` in [force-push](https://github.com/nextstrain/augur/compare/2582221ee2b729c4399a86a6f4f82d1b3bb37dd4..6e6e20d4eb6a9cf555ea122e5a79ae53b20bb342).

Reply:
Using DSN conventions sounds like a good idea if/when we want to support other database implementations. However, I'd like to limit the scope of this PR. How about only supporting SQLite3 database filepaths for `--db`/`--output-db` for now?

If/when we consider adding support for a remote database server (e.g. PostgreSQL), we should consider using DSNs to overload the `db` arguments.

If/when we consider adding support for another file-based database implementation (e.g. [DuckDB](https://duckdb.org/)), we should consider DSNs as well as adding the ability to decide which database implementation to use for filepaths.